error.go

```diff
@@ -27,12 +27,85 @@ package temporal
+		  what kind of error it was and take actions based on it. The details is encoded payload which workflow code could extract
+		  to strong typed variable. Workflow code needs to know what the types of the encoded details are before extracting them.
+		- If activity implementation returns errors other than from NewApplicationError() API. In this case GetOriginalType()
+		  will return orginal type of an error represented as string. Workflow code could check this type to determine what kind of error it was
+		  and take actions based on the type. These errors are retryable by default, unless error type is specified in retry policy.
+2) *CanceledError:
+	If activity was canceled, internal error will be an instance of *CanceledError. When activity cancels itself by
+	returning NewCancelError() it would supply optional details which could be extracted by workflow code.
+3) *TimeoutError:
+	If activity was timed out (several timeout types), internal error will be an instance of *TimeoutError. The err contains
+	details about what type of timeout it was.
+4) *PanicError:
+	If activity code panic while executing, temporal activity worker will report it as activity failure to temporal server.
+	The SDK will present that failure as *PanicError. The err contains a string	representation of the panic message and
+	the call stack when panic was happen.
+
+Workflow code could handle errors based on different types of error. Below is sample code of how error handling looks like.
+
+_, err := workflow.ExecuteActivity(ctx, MyActivity, ...).Get(nil)
+if err != nil {
+	var applicationErr *ApplicationError
+	if errors.As(err, &applicationError) {
+		// handle activity errors (created via NewApplicationError() API)
+		if !applicationErr.NonRetryable() {
+			// manually retry activity
+		}
+		var detailMsg string // assuming activity return error by NewApplicationError("message", true, "string details")
+		applicationErr.Details(&detailMsg) // extract strong typed details
+
+		// handle activity errors (errors created other than using NewApplicationError() API)
+		switch err.OriginalType() {
+		case "CustomErrTypeA":
+			// handle CustomErrTypeA
+		case CustomErrTypeB:
+			// handle CustomErrTypeB
+		default:
+			// newer version of activity could return new errors that workflow was not aware of.
+		}
+	}
+
+	var canceledErr *CanceledError
+	if errors.As(err, &canceledErr) {
+		// handle cancellation
+	}
+
+	var timeoutErr *TimeoutError
+	if errors.As(err, &timeoutErr) {
+		// handle timeout, could check timeout type by timeoutErr.TimeoutType()
+	}
+
+	var panicErr *PanicError
+	if errors.As(err, &panicErr) {
+		// handle panic, message and stack trace are available by panicErr.Error() and panicErr.StackTrace()
+	}
+}
+
+Errors from child workflow should be handled in a similar way, except that instance of *ChildWorkflowExecutionError is returned to
+workflow code. It will contains *ActivityTaskError, which in turn will contains on of the errors above.
```

Comment:
This is confusing. It might contain for example an application error that a child workflow returned directly